python/examples/minimal_build/build_venv.sh

```diff
@@ -35,6 +35,7 @@ source $WORKDIR/venv/bin/activate
 git config --global --add safe.directory $ARROW_ROOT
 
 pip install -r $ARROW_ROOT/python/requirements-build.txt
+pip install wheel
```

Comment:
Ah, I see we actually _do_ list `wheel` as a build requirement ;) But I think nowadays the recommendation is not list `wheel` explicitly. 
While I can't find an explicit reference for this, the original PEP text included `wheel` in the simple example (https://peps.python.org/pep-0518/#build-system-table), but the examples for using setuptools in the packaging guide or on the setuptools documentation (https://setuptools.pypa.io/en/latest/userguide/quickstart.html#basic-use) no longer include `wheel`.

So I _think_ we should remove `wheel` from `build-system.requires`

Reply:
Done

Reply:
@jorisvandenbossche It seems wheel is required otherwise we get the following failure:
```
 Processing /arrow/python
  Added file:///arrow/python to build tracker '/tmp/pip-req-tracker-grmxa_d3'
  Created temporary directory: /tmp/pip-modern-metadata-sclz_5km
  Preparing metadata (pyproject.toml): started
  Running command Preparing metadata (pyproject.toml)
  <string>:34: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  running dist_info
  creating /tmp/pip-modern-metadata-sclz_5km/pyarrow.egg-info
  writing /tmp/pip-modern-metadata-sclz_5km/pyarrow.egg-info/PKG-INFO
  writing dependency_links to /tmp/pip-modern-metadata-sclz_5km/pyarrow.egg-info/dependency_links.txt
  writing requirements to /tmp/pip-modern-metadata-sclz_5km/pyarrow.egg-info/requires.txt
  writing top-level names to /tmp/pip-modern-metadata-sclz_5km/pyarrow.egg-info/top_level.txt
  writing manifest file '/tmp/pip-modern-metadata-sclz_5km/pyarrow.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no previously-included files matching '*.so' found anywhere in distribution
  warning: no previously-included files matching '*.pyc' found anywhere in distribution
  warning: no previously-included files matching '*~' found anywhere in distribution
  warning: no previously-included files matching '#*' found anywhere in distribution
  warning: no previously-included files matching '.DS_Store' found anywhere in distribution
  no previously-included directories found matching '.asv'
  adding license file '../LICENSE.txt'
  adding license file '../NOTICE.txt'
  writing manifest file '/tmp/pip-modern-metadata-sclz_5km/pyarrow.egg-info/SOURCES.txt'
  creating '/tmp/pip-modern-metadata-sclz_5km/pyarrow-17.0.0.dev252+g50a35c1b1.dist-info'
  error: invalid command 'bdist_wheel'
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  ```
  See: https://github.com/ursacomputing/crossbow/actions/runs/9348063637/job/25726407074

Reply:
Ah, OK, diving in a bit more to understand what is happening: this is because we are using `--no-build-isolation` here. So it seems that setuptools will (in normal, isolated mode) automatically inject `wheel` as a dependency when building a wheel, but so that doesn't happen in non-isolated mode.

So it's still a good change to exlude `wheel` from the build requirements in pyproject.toml, but then we need to add it to the environments that use `--no-build-isolation`.

Note that explains this a bit in the setuptools docs (https://setuptools.pypa.io/en/latest/userguide/quickstart.html#basic-use):

>    Historically this documentation has unnecessarily listed ``wheel``
   in the ``requires`` list, and many projects still do that. This is
   not recommended, as the backend no longer requires the ``wheel``
   package, and listing it explicitly causes it to be unnecessarily
   required for source distribution builds.
   You should only include ``wheel`` in ``requires`` if you need to explicitly
   access it during build time (e.g. if your project needs a ``setup.py``
   script that imports ``wheel``).

Reply:
> So it's still a good change to exlude `wheel` from the build requirements in pyproject.toml,

I may have missed something but is it really useful to exclude it from the build requirements? It's a pure Python package with zero non-trivial dependency, AFAICT.

Reply:
For example:
```console
$ pip list
Package    Version
---------- -------
pip        24.0
setuptools 59.6.0
$ pip install wheel
Collecting wheel
  Using cached wheel-0.43.0-py3-none-any.whl.metadata (2.2 kB)
Using cached wheel-0.43.0-py3-none-any.whl (65 kB)
Installing collected packages: wheel
Successfully installed wheel-0.43.0
```

Reply:
That's the recommendation from setuptools ..
(I agree it shouldn't hurt given it's an easy to install package, although I also noticed in the setuptools tracker that the next version of setuptools will actually remove `wheel` usage all together)